src/Bicep.Core/Parsing/TokenReader.cs

```diff
@@ -32,6 +32,12 @@ public Token Peek()
             return Tokens[Position];
         }
 
+        public Token? PeekAhead(int charCount = 1)
```

Comment:
Why not just have a single Peek function which takes an optional charCount? (see `Peek()` implementation in SlidingTextWindow.cs)

Reply:
They have different signatures. Peek returns a non-null `Token` and PeekAhead returns `Token?` One option would be to make `PeekAhead` lie and return EOF token when the specified position is out of bounds. Thoughts?